src/Perun/Error.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Module describing the offchain errors the Perun-Contract
+-- implementation can emit.
+module Perun.Error
+  ( PerunError (..),
+    FindChannelException (..),
+    AsPerunError (..),
+  )
+where
+
+import Control.Lens
+import Data.Aeson
+import GHC.Generics (Generic)
+import Plutus.Contract
+
+data PerunError
+  = -- | Generic ContractError, encapsulated here s.t. it can be caught
+    -- together with Perun specific contract errors.
+    PerunContractError !ContractError
+  | -- | Thrown when a party has balances remaining within (0, minAda).
+    InsufficientMinimumAdaFundingError
```

Comment:
Better name imo:
`InsufficientMinimumAdaBalanceError`
(Funding implies to me that it only occurs "in the process of funding the channel", but this is also thrown when we e.g. try to dispute with a new signed state that has any balance in `(0, minAda)`)

Reply:
True!